### PR TITLE
Dependency Uptake: CLI - part 1

### PR DESCRIPTION
**Pull request by ghost**
**Handling of the base URL for asset download with the requisite 'key'**

_build/AzureInfo.props_
'PB_AssetRootUrl' overrides the normal URL download root: "https://dotnetcli.azureedge.net/dotnet/" - 'CoreSetupBlobRootUrl'
'CoreSetupBlobAccessTokenParam' is an empty string until 'PB_AssetRootAccessTokenSuffix' is specified
"TransportFeed" should be "BlobFeed"

_dir.props_ 
The import of 'AzureInfo.props' needs to be sequenced before 'Bundled*.props'

_build/Publish.targets_ 
build/publish/PublishNupkgToBlobFeed.targets 
"TransportFeed" should be "BlobFeed"

### CODE REVIEW

File: build/compile/LzmaArchive.targets [hunk 1]
```diff
@@ -2,19 +2,17 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
     <PropertyGroup>
-      <!-- In the event of a non-anonymous storage account used for core-setup bits, support appending a SAS token to download files -->
-      <CoreSetupBlobAccessTokenParam Condition=" '$(CoreSetupBlobAccessToken)' != '' ">?$(CoreSetupBlobAccessToken)</CoreSetupBlobAccessTokenParam>
       <FinalArchive>$(SdkOutputDirectory)/nuGetPackagesArchive.lzma</FinalArchive>
       <NugetPackagesArchiveName>nuGetPackagesArchive.$(AspNetCoreRuntimePackageFlavor).lzma</NugetPackagesArchiveName>
       <IntermediateArchive>$(IntermediateDirectory)/$(NugetPackagesArchiveName)</IntermediateArchive>
-      <NugetPackagesArchiveBlobUrl>$(AspNetCoreRuntimeInstallerBlobRootUrl)/$(NugetPackagesArchiveName)$(CoreSetupBlobAccessTokenParam)</NugetPackagesArchiveBlobUrl>
+      <NugetPackagesArchiveBlobUrl>$(AspNetCoreRuntimeInstallerBlobRootUrl)/$(NugetPackagesArchiveName)</NugetPackagesArchiveBlobUrl>
```
Comment (livarcocc) [inline build/compile/LzmaArchive.targets]: Why isn't the token needed here anymore?

File: scripts/dockerrun.sh [hunk 2]
```diff
@@ -119,20 +119,14 @@ docker run $INTERACTIVE -t --rm --sig-proxy=true \
     --name $DOTNET_BUILD_CONTAINER_NAME \
     -v $DOCKER_HOST_SHARE_DIR:/opt/code \
     -e CHANNEL \
-    -e CONNECTION_STRING \
```
Comment (livarcocc) [inline scripts/dockerrun.sh]: Why is it we don't need these anymore? In the very least, I would expect we need the access token.

File: build/AzureInfo.props [hunk 3]
```diff
@@ -1,6 +1,10 @@
 <Project>
   <PropertyGroup>
     <Product>Sdk</Product>
+    <AssetDownloadRootUrl>$(PB_AssetRootUrl)</AssetDownloadRootUrl>
+    <AssetDownloadRootUrl Condition="'$(AssetDownloadRootUrl)' == ''">https://dotnetcli.azureedge.net</AssetDownloadRootUrl>
+    <PrivateBlobAccessToken>$(PB_AssetRootAccessTokenSuffix)</PrivateBlobAccessToken>
+    <PrivateBlobAccessTokenParam Condition="'$(PrivateBlobAccessToken)' != ''">?$(PrivateBlobAccessToken)</PrivateBlobAccessTokenParam>
```
Comment (livarcocc) [inline build/AzureInfo.props]: I think you should keep the PrivateBlobAccessTokenParam settable through CoreSetupBlobAccessTokenParam. As it is, you are breaking the MSRC build definitions.

File: scripts/dockerrun.sh (hunk 2 above)
Comment (johnbeisner) [reply scripts/dockerrun.sh]: The access token is passed explicitly as a msbuild 'secret' parameter to whatever step needs it - my understanding is that if it is passed explicitly; defining it again in the environment is redundant.

File: build/compile/LzmaArchive.targets (hunk 1 above)
Comment (johnbeisner) [reply build/compile/LzmaArchive.targets]: I moved it down to the call; standardizing it with all the other calls. See: 'Installer.DEB.targets'; 'BackwardsCompatibilityRuntimes.props'; 'BundledRuntimes.props'; 'Installer.RPM.targets'; et al

File: build/AzureInfo.props [hunk 4]
```diff
@@ -1,6 +1,10 @@
 <Project>
   <PropertyGroup>
     <Product>Sdk</Product>
+    <CoreSetupBlobRootUrl>$(PB_AssetRootUrl)</CoreSetupBlobRootUrl>
+    <CoreSetupBlobRootUrl Condition="'$(CoreSetupBlobRootUrl)' == ''">https://dotnetcli.azureedge.net/dotnet/</CoreSetupBlobRootUrl>
+    <CoreSetupBlobAccessToken>$(PB_AssetRootAccessTokenSuffix)</CoreSetupBlobAccessToken>
```
Comment (livarcocc) [inline build/AzureInfo.props]: Should this be condition on CoreSetupBlobAccessToken being empty?

File: build/BundledRuntimes.props [hunk 5]
```diff
@@ -27,8 +27,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <CoreSetupBlobRootUrl Condition="'$(CoreSetupBlobRootUrl)' == ''">https://dotnetcli.azureedge.net/dotnet/</CoreSetupBlobRootUrl>
-    <CoreSetupBlobAccessTokenParam Condition=" '$(CoreSetupBlobAccessToken)' != '' ">?$(CoreSetupBlobAccessToken)</CoreSetupBlobAccessTokenParam>
+    <CoreSetupBlobRootUrl Condition="'$(CoreSetupBlobRootUrl)' == ''">$(AssetDownloadRootUrl)/dotnet/</CoreSetupBlobRootUrl>
```
Comment (livarcocc) [inline build/BundledRuntimes.props]: Is this right?

File: scripts/dockerrun.sh (hunk 2 above)
Comment (livarcocc) [reply scripts/dockerrun.sh]: But without this, it won't flow into the docker container. Which is why all of these are here. When the build with docker, the env variables are set in the hosting machine, than we start a docker VM where the actual build happens. Not everything is passed down, only what is defined here.

Comment (johnbeisner) [reply scripts/dockerrun.sh]: Reverting...

File: build/BundledRuntimes.props (hunk 5 above)
Comment (johnbeisner) [reply build/BundledRuntimes.props]: Line 30 should not be there; this is defined in 'AzureInfo.props'

File: build/AzureInfo.props (hunk 4 above)
Comment (johnbeisner) [reply build/AzureInfo.props]: Fixed...

File: build/AzureInfo.props [hunk 6]
```diff
@@ -1,6 +1,10 @@
 <Project>
   <PropertyGroup>
     <Product>Sdk</Product>
+    <CoreSetupBlobRootUrl Condition="'$(PB_AssetRootUrl)' != ''">$(PB_AssetRootUrl)</CoreSetupBlobRootUrl>
```
Comment (mmitche) [inline build/AzureInfo.props]: Could would we not simply set /p:CoreSetupBlobRootUrl=$(PB_AssetRootUrl) on the msbuild line and remove this code here instead?

File: build/publish/PublishNupkgToBlobFeed.targets [hunk 7]
```diff
@@ -7,13 +7,13 @@
     <NupkgsForPublishing Include="$(PackagesDirectory)/Microsoft.DotNet.MSBuildSdkResolver.*.nupkg" />
   </ItemGroup>
 
-  <Target Name="PublishNupkgToTransportFeed" Condition=" '$(PUBLISH_NUPKG_TO_TRANSPORT_FEED)' == 'true' AND '$(OS)' == 'Windows_NT' And '$(Architecture)' == 'x64' " >
-    <Error Condition="'$(TransportFeedCloudDropAccessToken)' == ''" Text="Missing property TransportFeedCloudDropAccessToken." />
-    <Error Condition="'$(TransportFeedExpectedURL)' == ''" Text="Missing property TransportFeedExpectedURL." />
+  <Target Name="PublishNupkgToBlobFeed" Condition=" '$(PUBLISH_NUPKG_TO_BLOB_FEED)' == 'true' AND '$(OS)' == 'Windows_NT' And '$(Architecture)' == 'x64' " >
```
Comment (mmitche) [inline build/publish/PublishNupkgToBlobFeed.targets]: Does PUBLISH_NUPKG_TO_BLOB_FEED hook up to PB_PublishType somewhere?

File: build/AzureInfo.props (hunk 6 above)
Comment (johnbeisner) [reply build/AzureInfo.props]: MSRC builds pass in 'CoreSetupBlobRootUrl'. See: https://devdiv.visualstudio.com/DevDiv/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=7080

That would result in setting the property to an empty string in the MSRC case.

File: build/publish/PublishNupkgToBlobFeed.targets (hunk 7 above)
Comment (johnbeisner) [reply build/publish/PublishNupkgToBlobFeed.targets]: Just before merge, I will need to change "PUBLISH_NUPKG_TO_TRANSPORT_FEED" to "PUBLISH_NUPKG_TO_BLOB_FEED" in the VSO build definitions. See: https://devdiv.visualstudio.com/DevDiv/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=6786

**Review by livarcocc — APPROVED** [review]

**Pull request merged**